### PR TITLE
java auto publish

### PR DESCRIPTION
**Pull request by stack72**
- [internal] Adding the work to auto-publish the java sdk
- Adding Java SDK generated files

This PR brings in #1829 as part of it to ensure the generation is correct

### CODE REVIEW

File: sdk/java/src/main/java/com/pulumi/azurenative/Utilities.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public static String getVersion() {
     }
 
     static {
-        var resourceName = "com/pulumi/azure-native/version.txt";
+        var resourceName = "com/pulumi/azurenative/version.txt";
```
Comment (pawelprazak) [inline sdk/java/src/main/java/com/pulumi/azurenative/Utilities.java:78]: hmm, need to make sure this file gets created properly by the `build.gradle`:
`sdk/java/src/resources/com/pulumi/azurenative/version.txt`

Like here: https://github.com/pulumi/pulumi-kubernetes/blob/cbc8c830eda8bd0fea78919d0f53ca7275a0086f/sdk/java/build.gradle#L68

**Comment by danielrbradley** [comment]
As discussed, let's build from the `schema-full.json` to get the API versions too. This is a build artifact and not committed, so we need to ensure it's there before the generate_java step.

This will massively increase the SDK size so we might want to not commit the Java SDK.

File: sdk/java/src/main/java/com/pulumi/azurenative/Utilities.java (hunk 1 above)
Comment (pawelprazak) [reply sdk/java/src/main/java/com/pulumi/azurenative/Utilities.java:78]: FYI: released in Java SDK `0.5.4`

**Comment by t0yv0** [comment]
AFAIK Java SDK currently won't be able to build from the FULL schema in any reasonable time (CPU time + memory). Last time we tried builds were not finishing in 30 min. We can reopen a ticket in pulumi/pulumi-java.

If it is possible somehow to keep releasing azure-native from the partial schema for now we'd really appreciate that.

**Pull request closed without merging**